### PR TITLE
expressions: use integer for time in arithmetic op if FSP = 0

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: expression/expressions/binop.go [hunk 1]
```diff
@@ -675,11 +675,21 @@ func (o *BinaryOperation) coerceArithmetic(a interface{}) (interface{}, error) {
...
+			return v.IntPart(), nil
+		}
+		return v, nil
 	case mysql.Duration:
-		// TODO: if duration has no precision, return int64
-		return x.ToNumber(), nil
+		// if duration has no precision, return int64
+		v := x.ToNumber()
+		if x.Fsp == 0 {
+			// Integer number 111111 can not overflow int64.
```
Comment (qiuyesuifeng) [inline expression/expressions/binop.go]: Seems 111111 cannot overflow?

Comment (siddontang) [inline expression/expressions/binop.go]: 111111 < math.Int64 of course

Comment (qiuyesuifeng) [inline expression/expressions/binop.go]: Got it.
But if make sure the ret value is valid, this comment is not needed?

**Comment by coocood** [comment]
LGTM

**Pull request merged**